test/unittests/libweb3core/overlaydb.cpp

```diff
@@ -20,22 +20,23 @@
  * OverlayDB tests.
  */
 
-#include <libdevcore/DBImpl.h>
 #include <libdevcore/OverlayDB.h>
+#include <libdevcore/DBFactory.h>
 #include <libdevcore/TransientDirectory.h>
 #include <test/tools/libtesteth/TestOutputHelper.h>
 #include <boost/test/unit_test.hpp>
 
 using namespace std;
 using namespace dev;
+using namespace db;
 using namespace dev::test;
 
 BOOST_FIXTURE_TEST_SUITE(OverlayDBTests, TestOutputHelperFixture)
 
 BOOST_AUTO_TEST_CASE(basicUsage)
 {
     TransientDirectory td;
-    std::unique_ptr<db::DBImpl> db(new db::DBImpl(td.path()));
+	std::unique_ptr<db::DatabaseFace> db = DBFactory::create(DatabaseKind::LevelDB, td.path());
```

Comment:
In all the tests in this file, I'd like to use `MemoryDB` instead of `LevelDB` and not use `TransientDirectory`, but this may go into separate PR.

Reply:
Got it, I'll take care of that in a separate PR

Reply:
Should we just create a copy of this file which uses MemoryDB since we presumably want test coverage for both the LevelDB and MemoryDB implementations (since LevelDB is the database actually used by users, at least until I get RocksDB online)?

Reply:
I don't think we need this. I believe unit tests by their nature should be as isolated as possible, so `OverlayDB` tests should test only the code in `OverlayDB`
(this is not quite the case here, since `MemoryDB` is used, so it's not ideal)

We could have additionally some set of tests checking how our `LevelDBDatabase` implementation works, but then `OverlayDB` shouldn't interfere with that.

Reply:
@gumb0 can you propose a code change and push it to this PR?

Reply:
I meant that we don't need additional test file as Nils proposed above, this code in this PR is fine